### DISCUSSION TITLE
How to install nvim-treesitter using :Rocks?

### DISCUSSION THREAD

**Discussion by craigmac** [lumen-oss/rocks.nvim#416] [Q&A] [answered]
I used to be able to do `:Rocks install nvim-treesitter` and in my rocks.toml it put `nvim-treesitter = "0.9.2"` but it seems now I can't install nvim-treesitter that way anymore? There's `nvim-treesitter-context`, `nvim-treesitter-legacy-api`, and `nvim-treesitter-textobjects` but no `nvim-treesitter`, what happened?

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave:

nvim-treesitter 0.9.2 is quite outdated and causes problems with the luarocks tree-sitter parsers, which is why we've removed it.

For now, you have two options:

- Install nvim-treesitter with rocks-git.nvim (not recommended)
- Install the tree-sitter-<lang> parsers you need (either manually or using rocks-treesitter.nvim)
  - For plugins that still depend on nvim-treesitter,
    we provide the `nvim-treesitter-legacy-api` rock, which is nvim-treesitter but patched not to include queries that could conflict with luarocks parsers.

**Reply by craigmac** [reply]
Thanks for the reply, I tried going with the nvim-treesitter-legacy-api rock, but ended up super frustrated with errors about uv_dlopen and some other things that I just don't have anymore time to look into. I think the whole nvim-treesitter compatibility with rocks.nvim needs to be spelled out much better in the readme, it seems overly complicated and fairly easy to break.

**Reply by mrcjkb** [reply]
> I think the whole nvim-treesitter compatibility with rocks.nvim needs to be spelled out much better in the readme

To be fair, rocks-treesitter.nvim does have a the following *large yellow warning notice*:

> [!WARNING]
>
> This module is **A WORK IN PROGRESS** and does not have a stable release yet.

And both rocks.nvim and rocks-treesitter.nvim have this one: 

> [!WARNING]
>
> - Tree-sitter and treesitter highlighting are an experimental feature of Neovim.
>   As with nvim-treesitter, please consider tree-sitter support with this plugin
>   experimental.

nvim-treesitter has such a warning too, it's just less prominent.

The `uv_dlopen` issue sounds like it's probably related to a C++ parser. Support for C++ parsers is deprecated. You could try the `scm` version of `nvim-treesitter-legacy-api`.

And it's not unique to rocks.nvim: https://github.com/search?q=repo%3Anvim-treesitter%2Fnvim-treesitter+uv_dlopen&type=issues

Regarding compatibility with nvim-treesitter: They were initially planning to go stable with nvim 0.10 but postponed it (apparently to nvim 0.11, but experience says that's probably not going to happen).
The fact that there are currently two different main branches (`master` and `main`) being maintained, neither of which have semver releases, makes the whole thing a huge PITA.

There is a tip box in the readme recommending not to use the nvim-treesitter with luarocks parsers.
And we plan on adding health checks for conflicts, etc.
But again, this is all still WIP and is prone to break whenever nvim-treesitter pushes a breaking change without having semver releases.
